contrib/epee/src/net_ssl.cpp

```diff
@@ -652,6 +654,44 @@ bool ssl_options_t::handshake(
   return true;
 }
 
+std::vector<std::uint8_t> convert_fingerprint(const boost::string_ref id)
+{
+  std::vector<std::uint8_t> out;
+  out.resize(id.size());
+  std::copy(id.begin(), id.end(), out.begin());
+  return out;
+}
+
+std::string get_ssl_fingerprint(const X509 *cert, const EVP_MD *fdig)
+{
+  CHECK_AND_ASSERT_THROW_MES(cert && fdig, "Pointer args to get_ssl_fingerprint cannot be null");
+
+  std::string md;
+  unsigned n = 0;
+  md.resize(EVP_MAX_MD_SIZE);
+  if (!X509_digest(cert, fdig, reinterpret_cast<std::uint8_t*>(&md[0]), &n))
+  {
+    const unsigned long ssl_err_val = static_cast<int>(ERR_get_error());
+    const boost::system::error_code ssl_err_code = boost::asio::error::ssl_errors(static_cast<int>(ssl_err_val));
+    MERROR("Failed to create SSL fingerprint: " << ERR_reason_error_string(ssl_err_val));
+    throw boost::system::system_error(ssl_err_code, ERR_reason_error_string(ssl_err_val));
+  }
+  md.resize(n);
+  return md;
+}
+
+std::string get_ssl_fingerprint(boost::asio::ssl::context &ssl_context, const EVP_MD *fdig)
+{
+  SSL_CTX *ctx = ssl_context.native_handle();
+  CHECK_AND_ASSERT_THROW_MES(ctx, "Failed to get SSL context");
+  
+  X509* ssl_cert = nullptr;
```

Comment:
Free with `X509_free` when done?

Reply:
The documentation for `SSL_get_certificate` states:

> Returned objects are owned by the SSL object and should not be freed by users of these functions.

So the `dflt_SSL` `unique_ptr` should cleanup the memory for this object.